firestore/report.go

```diff
@@ -0,0 +1,76 @@
+package firestore
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/firestore"
+
+	"github.com/benchttp/server/benchttp"
+)
+
+// Ensure service implements interface.
+var _ benchttp.ReportService = (*ReportService)(nil)
+
+type ReportService struct {
+	client       *firestore.Client
+	collectionID string
+}
+
+// NewReportService returns a new repository wrapping a
+// Firestore client. The internal client uses the given project.
+// The repository only uses collectionID as the collection to
+// create documents to and retrieve documents from.
+func NewReportService(ctx context.Context, projectID, collectionID string) (*ReportService, error) {
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, err
+	}
+
+	s := &ReportService{
+		client:       client,
+		collectionID: collectionID,
+	}
+	return s, nil
+}
+
+// CloseClient calls the Close method of the internal client
+// of the ReportService.
+func (s ReportService) CloseClient() {
+	s.client.Close()
+}
```

Comment:
[question] This is not used, what should we do about it?

[nitpick] I'd keep the implementation details to the implementation itself, considering the underlying client is private.
```suggestion
// Close closes the connection to Firestore client.
func (s ReportService) Close() error {
	return s.client.Close()
}
```